proto/src/lib.rs

```diff
@@ -23,6 +23,9 @@ pub mod serializers;
 mod protobuf;
 pub use protobuf::Protobuf;
 
+mod error;
+pub use error::ProtoError;
```

Comment:
What's the motivation for pulling `ProtoError` into `penumbra_proto` from the `transaction` crate?  I'm not 100% sure, but my intuition is that we probably don't want it here (if we want it at all -- it might be better to remove it entirely), since the `ProtoError` has errors relating to the conversion between a proto type and a domain type, and the domain types don't live in the `proto` crate.

Reply:
There was a circular dependency I was dealing with, but it's no longer necessary. Agreed about removing it -- there was a comment about just replacing it with `anyhow`. Is the idea that we don't want to use error enums at all in the codebase?